sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -4031,6 +4031,28 @@ def hpat_pandas_series_min_impl(self, axis=None, skipna=None, level=None, numeri
     return hpat_pandas_series_min_impl
 
 
+@numba.extending.register_jitable(parallel=True)
+def numpy_nanmax(a):
+    length = len(a)
+
+    # set first element
+    result = numpy.nan
+    for i in range(length):
+        value = a[i]
+        if not numpy.isnan(value):
+            result = value
+            break
+    else:
+        return result
```

Comment:
doesn't looks good. In case if all values are `nan` this function wouldn't scale at all.

Reply:
There are utility functions in numpy which returns maximum/minimum values for different types:
```
min_float64 = np.finfo('float64').min
max_float64 = np.finfo('float64').max
min_int64 = np.iinfo('int64').min
max_int64 = np.iinfo('int64').max
```
And for the rest. We need to implement utility function which would return min/max value based on variable type.
Please also note that we can't the same value for all int types - result type should be of the same type as array. I think so.

For floats we should use `nan`. I think